onadata/apps/api/viewsets/v2/imports_viewset.py

```diff
@@ -0,0 +1,214 @@
+        queryset = self.get_queryset()
+        obj = get_object_or_404(queryset, pk=pk)
+        self.check_object_permissions(self.request, obj)
+        return obj
+
+    def _get_active_tasks(self, xform: XForm) -> str:
+        """
+        Utility function to retrieve the active tasks of an XForm
+        """
+        return get_active_tasks(self.task_names, xform)
+
+    # pylint: disable=invalid-name
+    def create(self, request, pk: int = None) -> Response:
+        """
+        Starts a new Import task for a given form; The route processes imports
+        asynchronously unless the `DISABLE_ASYNCHRONOUS_IMPORTS` setting
+        is set to false.
+
+        Curl example:
+
+        $ curl -X POST "http://<domain>/api/v2/imports/<xform_id>"
+
+        Supported Query Parameters:
+
+        - [Optional] overwrite: bool = Whether the server should permanently
+          delete the data currently available on the form then reimport the
+          data using the csv_file/xls_file sent with the request.
+
+        Required Request Arguements:
+
+        - csv_file: file = A CSV File containing the forms data
+        - xls_file: file = An XLSX file containing the forms data
+
+        Possible Response status codes:
+
+        - 202 Accepted: Server has successfully accepted your request for data
+          import and has queued the task
+        - 200 Ok: Server has successfully imported your data to the form;
+          Only returned when asynchronous imports are disabled
+        - 400 Bad Request: Request has been refused due to incorrect/missing
+          csv_file or xls_file file
+        - 403 Forbidden: The request was valid but the server refused to
+          process it. An explanation on why it was refused can be found
+          in the JSON Response
+        - 401 Unauthorized: The request has been refused due to missing authentication
+        """
+        xform = self.get_object(pk)
+        resp = {}
+        csv_file = request.FILES.get("csv_file", None)
+        xls_file = request.FILES.get("xls_file", None)
+        status_code = status.HTTP_400_BAD_REQUEST
+
+        if csv_file is None and xls_file is None:
+            resp.update({"error": "csv_file and xls_file field empty"})
+        elif xls_file and xls_file.name.split(".")[-1] not in XLS_EXTENSIONS:
+            resp.update({"error": "xls_file not an excel file"})
+        elif csv_file and csv_file.name.split(".")[-1] != CSV_EXTENSION:
+            resp.update({"error": "csv_file not a csv file"})
+        else:
+            if xls_file and xls_file.name.split(".")[-1] in XLS_EXTENSIONS:
+                csv_file = submission_xls_to_csv(xls_file)
+
+            overwrite = request.query_params.get("overwrite")
+            overwrite = (overwrite.lower() == "true" if isinstance(
+                overwrite, str) else overwrite)
+
+            # Block imports from running when an overwrite is ongoing
+            active_tasks = self._get_active_tasks(xform)
```

Comment:
Why not `get_active_tasks(self.task_names, xform)` it is still clean and there is nothing special about the `_get_active_tasks()` that requires it to be a class method.

Reply:
Will make this change